lib/ZcapClient.js

```diff
@@ -0,0 +1,228 @@
+/*!
+ * Copyright (c) 2020 Digital Bazaar, Inc. All rights reserved.
+ */
+import {DEFAULT_HEADERS, httpClient} from '@digitalbazaar/http-client';
+import {SECURITY_CONTEXT_V2_URL} from 'jsonld-signatures';
+import {CapabilityDelegation} from 'ocapld';
+import {generateId} from 'bnid';
+import jsigs from 'jsonld-signatures';
+import {signCapabilityInvocation} from 'http-signature-zcap-invoke';
+const {Ed25519Signature2018} = jsigs.suites;
+
+export class ZcapClient {
+  /**
+   * An object that can make HTTPS requests such as node's https.Agent or ky.
+   *
+   * @typedef {object} httpsAgent
+   * @see https://nodejs.org/api/https.html#https_class_https_agent
+   *
+   * Creates a new ZcapClient instance that can be used to perform
+   * Authorization Capability (ZCAP) requests against HTTP URLs.
```

Comment:
This is confusing. Should this come before the `HttpsAgent` typedef? It seems to be the opening statement about what the constructor does.

Reply:
Fixed in 447ce41564ddf248babbefb3f5fb622407fbf713.